File: Overlord.Core/Overlord.Core/Models/AIModels.cs
namespace Overlord.Core.Models;

/// <summary>
/// AI personality archetypes that influence decision-making.
/// </summary>
public enum AIPersonality
{
    /// <summary>
    /// Aggressive warmonger - prioritizes military over economy, attacks early.
    /// Commander Kratos.
    /// </summary>
    Aggressive,

    /// <summary>
    /// Defensive turtle - prioritizes defense, rarely attacks, strong late-game.
    /// Overseer Aegis.
    /// </summary>
    Defensive,

    /// <summary>
    /// Economic expansionist - prioritizes resource production, weak early-game.
    /// Magistrate Midas.
    /// </summary>
    Economic,

    /// <summary>
    /// Balanced tactical - adapts to player actions, no extreme tendencies.
    /// General Nexus.
    /// </summary>
    Balanced
}

/// <summary>
/// AI difficulty levels that apply bonuses/penalties.
/// </summary>
public enum AIDifficulty
{
    /// <summary>
    /// Easy: -20% resource production, -20% military strength, cautious aggression.
    /// </summary>
    Easy,

    /// <summary>
    /// Normal: No bonuses/penalties, balanced aggression.
    /// </summary>
    Normal,

    /// <summary>
    /// Hard: +20% resource production, +20% military strength, aggressive expansion.
    /// </summary>
    Hard
}

/// <summary>
/// AI personality configuration with modifiers.
/// </summary>
public class AIPersonalityConfig
{
    public AIPersonality Personality { get; set; }
    public string Name { get; set; } = string.Empty;
    public string Quote { get; set; } = string.Empty;

    /// <summary>
    /// Aggression modifier: +0.5 = more aggressive, -0.5 = more defensive.
    /// </summary>
    public float AggressionModifier { get; set; }

    /// <summary>
    /// Economic modifier: +0.5 = prioritize economy, -0.3 = deprioritize economy.
    /// </summary>
    public float EconomicModifier { get; set; }

    /// <summary>
    /// Defense modifier: +0.4 = prioritize defense, 0 = normal.
    /// </summary>
    public float DefenseModifier { get; set; }

    /// <summary>
    /// Gets personality configuration for a given type.
    /// </summary>
    public static AIPersonalityConfig GetConfig(AIPersonality personality)
    {
        return personality switch
        {
            AIPersonality.Aggressive => new AIPersonalityConfig
            {
                Personality = AIPersonality.Aggressive,
                Name = "Commander Kratos",
                Quote = "Strength through conquest! Your planets will fall.",
                AggressionModifier = 0.5f,
                EconomicModifier = -0.3f,
                DefenseModifier = 0.0f
            },
            AIPersonality.Defensive => new AIPersonalityConfig
            {
                Personality = AIPersonality.Defensive,
                Name = "Overseer Aegis",
                Quote = "Patience is the strongest fortress. I can wait.",
                AggressionModifier = -0.5f,
                EconomicModifier = 0.0f,
                DefenseModifier = 0.4f
            },
            AIPersonality.Economic => new AIPersonalityConfig
            {
                Personality = AIPersonality.Economic,
                Name = "Magistrate Midas",
                Quote = "Credits are power. My wealth will crush you.",
                AggressionModifier = -0.3f,
                EconomicModifier = 0.5f,
                DefenseModifier = 0.0f
            },
            AIPersonality.Balanced => new AIPersonalityConfig
            {
                Personality = AIPersonality.Balanced,
                Name = "General Nexus",
                Quote = "Strategy, not emotion, wins wars.",
                AggressionModifier = 0.0f,
                EconomicModifier = 0.0f,
                DefenseModifier = 0.0f
            },
            _ => throw new ArgumentException($"Unknown personality: {personality}")
        };
    }
}

/// <summary>
/// AI game state assessment result.
/// </summary>
public class AIAssessment
{
    /// <summary>
    /// Threat level: Player military strength ÷ AI military strength.
    /// <![CDATA[< 0.5 = AI dominant, > 1.5 = Player dominant]]>
    /// </summary>
    public float ThreatLevel { get; set; }

    /// <summary>
    /// Economic strength ratio: AI resources ÷ Player resources.
    /// </summary>
    public float EconomicStrength { get; set; }

    /// <summary>
    /// Number of AI-owned planets.
    /// </summary>
    public int AIPlanets { get; set; }

    /// <summary>
    /// Number of player-owned planets.
    /// </summary>
    public int PlayerPlanets { get; set; }

    /// <summary>
    /// True if player has fleet at AI planet (under attack).
    /// </summary>
    public bool UnderAttack { get; set; }

    /// <summary>
    /// True if AI has military advantage and can attack.
    /// </summary>
    public bool CanAttack { get; set; }
}

File: Overlord.Phaser/src/core/models/AIModels.ts
import { AIPersonality } from './Enums';

/**
 * AI personality configuration with modifiers.
 */
export class AIPersonalityConfig {
  public personality: AIPersonality = AIPersonality.Balanced;
  public name: string = '';
  public quote: string = '';

  /**
   * Aggression modifier: +0.5 = more aggressive, -0.5 = more defensive.
   */
  public aggressionModifier: number = 0;

  /**
   * Economic modifier: +0.5 = prioritize economy, -0.3 = deprioritize economy.
   */
  public economicModifier: number = 0;

  /**
   * Defense modifier: +0.4 = prioritize defense, 0 = normal.
   */
  public defenseModifier: number = 0;

  /**
   * Gets personality configuration for a given type.
   */
  public static getConfig(personality: AIPersonality): AIPersonalityConfig {
    const config = new AIPersonalityConfig();

    switch (personality) {
      case AIPersonality.Aggressive:
        config.personality = AIPersonality.Aggressive;
        config.name = 'Commander Kratos';
        config.quote = 'Strength through conquest! Your planets will fall.';
        config.aggressionModifier = 0.5;
        config.economicModifier = -0.3;
        config.defenseModifier = 0.0;
        break;

      case AIPersonality.Defensive:
        config.personality = AIPersonality.Defensive;
        config.name = 'Overseer Aegis';
        config.quote = 'Patience is the strongest fortress. I can wait.';
        config.aggressionModifier = -0.5;
        config.economicModifier = 0.0;
        config.defenseModifier = 0.4;
        break;

      case AIPersonality.Economic:
        config.personality = AIPersonality.Economic;
        config.name = 'Magistrate Midas';
        config.quote = 'Credits are power. My wealth will crush you.';
        config.aggressionModifier = -0.3;
        config.economicModifier = 0.5;
        config.defenseModifier = 0.0;
        break;

      case AIPersonality.Balanced:
        config.personality = AIPersonality.Balanced;
        config.name = 'General Nexus';
        config.quote = 'Strategy, not emotion, wins wars.';
        config.aggressionModifier = 0.0;
        config.economicModifier = 0.0;
        config.defenseModifier = 0.0;
        break;

      default:
        throw new Error(`Unknown personality: ${personality}`);
    }

    return config;
  }
}

/**
 * AI game state assessment result.
 */
export class AIAssessment {
  /**
   * Threat level: Player military strength ÷ AI military strength.
   * < 0.5 = AI dominant, > 1.5 = Player dominant
   */
  public threatLevel: number = 0;

  /**
   * Economic strength ratio: AI resources ÷ Player resources.
   */
  public economicStrength: number = 0;

  /**
   * Number of AI-owned planets.
   */
  public aiPlanets: number = 0;

  /**
   * Number of player-owned planets.
   */
  public playerPlanets: number = 0;

  /**
   * True if player has fleet at AI planet (under attack).
   */
  public underAttack: boolean = false;

  /**
   * True if AI has military advantage and can attack.
   */
  public canAttack: boolean = false;
}
